packages/components/src/components/autocomplete/autocomplete.tsx

```diff
@@ -16,21 +42,19 @@ const KEY_CODE = {
     KEYUP: 38,
 };
 
-const getFilteredItems = (val, list, should_filter_by_char) => {
-    const is_string_array = list.length && typeof list[0] === 'string';
-
+const getFilteredItems = (val: string, list: TItem[], should_filter_by_char = false) => {
     if (should_filter_by_char) {
         return list.filter(item =>
-            is_string_array ? matchStringByChar(item, val) : matchStringByChar(item.text, val)
+            typeof item === 'string' ? matchStringByChar(item, val) : matchStringByChar(item.text || '', val)
         );
     }
     return list.filter(item =>
-        is_string_array
+        typeof item === 'string'
             ? getEnglishCharacters(item).toLowerCase().includes(val) || item.toLowerCase().includes(val)
-            : getEnglishCharacters(item.text).toLowerCase().includes(val) || item.text.toLowerCase().includes(val)
+            : getEnglishCharacters(item.text).toLowerCase().includes(val) || item?.text?.toLowerCase().includes(val)
```

Comment:
I see in line 48 we have `item.text || "" ` but here we are not doing the same thing when we pass `item.text `to `getEnglishCharacters` . don't we need to do the same thing here as well?